### PR TITLE
Remove archival code

### PR DESCRIPTION
**Pull request by Gnnng**
## Description

The archival mode is removed from this PR. As a result, mongo driver is no longer needed.

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [ ] local machine test

- [x] small-scale cloud test (commit: 9e81a62)

### CODE REVIEW

File: src/libLookup/Lookup.cpp [hunk 1]
```diff
@@ -1592,15 +1583,6 @@ bool Lookup::ProcessSetDSBlockFromSeed(const bytes& message,
         continue;
       }
       m_mediator.m_dsBlockChain.AddBlock(dsblock);
-      // Store DS Block to disk
-      if (!ARCHIVAL_NODE) {
```
Comment (KaustubhShamshery) [inline src/libLookup/Lookup.cpp]: the block should be stored if it is not archival node

File: src/libLookup/Lookup.cpp [hunk 2]
```diff
@@ -1738,25 +1720,6 @@ void Lookup::CommitTxBlocks(const vector<TxBlock>& txBlocks) {
     LogTxBlock(txBlock, m_mediator.m_currentEpochNum);
 
     m_mediator.m_node->AddBlock(txBlock);
-
-    // Store Tx Block to disk
-    if (!ARCHIVAL_NODE) {
```
Comment (KaustubhShamshery) [inline src/libLookup/Lookup.cpp]: same

Comment (KaustubhShamshery) [inline src/libLookup/Lookup.cpp]: keep the !ARCHIVAL_NODE part

File: src/libLookup/Lookup.cpp (hunk 1 above)
Comment (Gnnng) [reply src/libLookup/Lookup.cpp]: added it back

File: src/libLookup/Lookup.cpp (hunk 2 above)
Comment (Gnnng) [reply src/libLookup/Lookup.cpp]: added it back

**Comment by mzhou2018** [comment]
Got this message while compiling..

c++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-5/README.Bugs> for instructions.
tests/Utils/CMakeFiles/Test_IPConverter.dir/build.make:62: recipe for target 'tests/Utils/CMakeFiles/Test_IPConverter.dir/Test_IPConverter.cpp.o' fail                                               ed
make[2]: *** [tests/Utils/CMakeFiles/Test_IPConverter.dir/Test_IPConverter.cpp.o] Error 4
CMakeFiles/Makefile2:4493: recipe for target 'tests/Utils/CMakeFiles/Test_IPConverter.dir/all' failed
make[1]: *** [tests/Utils/CMakeFiles/Test_IPConverter.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....

^Ctests/Utils/CMakeFiles/Test_DataConversion.dir/build.make:62: recipe for target 'tests/Utils/CMakeFiles/Test_DataConversion.dir/Test_DataConversion.                                               cpp.o' failed
make[2]: *** [tests/Utils/CMakeFiles/Test_DataConversion.dir/Test_DataConversion.cpp.o] Interrupt
tests/Utils/CMakeFiles/Test_Logger3.dir/build.make:62: recipe for target 'tests/Utils/CMakeFiles/Test_Logger3.dir/Test_Logger3.cpp.o' failed
make[2]: *** [tests/Utils/CMakeFiles/Test_Logger3.dir/Test_Logger3.cpp.o] Interrupt
CMakeFiles/Makefile2:4607: recipe for target 'tests/Utils/CMakeFiles/Test_DataConversion.dir/all' failed
make[1]: *** [tests/Utils/CMakeFiles/Test_DataConversion.dir/all] Interrupt
tests/Utils/CMakeFiles/Test_SWInfo.dir/build.make:62: recipe for target 'tests/Utils/CMakeFiles/Test_SWInfo.dir/Test_SWInfo.cpp.o' failed
make[2]: *** [tests/Utils/CMakeFiles/Test_SWInfo.dir/Test_SWInfo.cpp.o] Interrupt
tests/Utils/CMakeFiles/Test_DetachedFunction.dir/build.make:62: recipe for target 'tests/Utils/CMakeFiles/Test_DetachedFunction.dir/Test_DetachedFunct                                               ion.cpp.o' failed
make[2]: *** [tests/Utils/CMakeFiles/Test_DetachedFunction.dir/Test_DetachedFunction.cpp.o] Interrupt
tests/Utils/CMakeFiles/Test_Serializable.dir/build.make:62: recipe for target 'tests/Utils/CMakeFiles/Test_Serializable.dir/Test_Serializable.cpp.o' f                                               ailed
make[2]: *** [tests/Utils/CMakeFiles/Test_Serializable.dir/Test_Serializable.cpp.o] Interrupt
tests/Utils/CMakeFiles/Test_SafeMath.dir/build.make:62: recipe for target 'tests/Utils/CMakeFiles/Test_SafeMath.dir/Test_SafeMath.cpp.o' failed
make[2]: *** [tests/Utils/CMakeFiles/Test_SafeMath.dir/Test_SafeMath.cpp.o] Interrupt
src/depends/libDatabase/CMakeFiles/Database.dir/build.make:62: recipe for target 'src/depends/libDatabase/CMakeFiles/Database.dir/LevelDB.cpp.o' faile                                               d
make[2]: *** [src/depends/libDatabase/CMakeFiles/Database.dir/LevelDB.cpp.o] Interrupt
src/depends/libDatabase/CMakeFiles/Database.dir/build.make:86: recipe for target 'src/depends/libDatabase/CMakeFiles/Database.dir/MemoryDB.cpp.o' fail                                               ed
make[2]: *** [src/depends/libDatabase/CMakeFiles/Database.dir/MemoryDB.cpp.o] Interrupt
src/depends/libDatabase/CMakeFiles/Database.dir/build.make:110: recipe for target 'src/depends/libDatabase/CMakeFiles/Database.dir/OverlayDB.cpp.o' fa                                               iled
make[2]: *** [src/depends/libDatabase/CMakeFiles/Database.dir/OverlayDB.cpp.o] Interrupt
tests/Data/CMakeFiles/Test_CircularArray.dir/build.make:62: recipe for target 'tests/Data/CMakeFiles/Test_CircularArray.dir/Test_CircularArray.cpp.o'                                                failed
make[2]: *** [tests/Data/CMakeFiles/Test_CircularArray.dir/Test_CircularArray.cpp.o] Interrupt
tests/Utils/CMakeFiles/Test_BoostBigNum.dir/build.make:62: recipe for target 'tests/Utils/CMakeFiles/Test_BoostBigNum.dir/Test_BoostBigNum.cpp.o' fail                                               ed
make[2]: *** [tests/Utils/CMakeFiles/Test_BoostBigNum.dir/Test_BoostBigNum.cpp.o] Interrupt
tests/Message/CMakeFiles/Test_Messenger_Compatibility.dir/build.make:94: recipe for target 'tests/Message/CMakeFiles/Test_Messenger_Compatibility.dir/                                               Test_Messenger_Compatibility.cpp.o' failed
make[2]: *** [tests/Message/CMakeFiles/Test_Messenger_Compatibility.dir/Test_Messenger_Compatibility.cpp.o] Interrupt
tests/Utils/CMakeFiles/Test_Logger1.dir/build.make:62: recipe for target 'tests/Utils/CMakeFiles/Test_Logger1.dir/Test_Logger1.cpp.o' failed
make[2]: *** [tests/Utils/CMakeFiles/Test_Logger1.dir/Test_Logger1.cpp.o] Interrupt
tests/Utils/CMakeFiles/Test_ShardSizeCalculator.dir/build.make:62: recipe for target 'tests/Utils/CMakeFiles/Test_ShardSizeCalculator.dir/Test_ShardSi                                               zeCalculator.cpp.o' failed
make[2]: *** [tests/Utils/CMakeFiles/Test_ShardSizeCalculator.dir/Test_ShardSizeCalculator.cpp.o] Interrupt
src/libCrypto/CMakeFiles/Crypto.dir/build.make:86: recipe for target 'src/libCrypto/CMakeFiles/Crypto.dir/MultiSig.cpp.o' failed
make[2]: *** [src/libCrypto/CMakeFiles/Crypto.dir/MultiSig.cpp.o] Interrupt
src/libCrypto/CMakeFiles/Crypto.dir/build.make:62: recipe for target 'src/libCrypto/CMakeFiles/Crypto.dir/Schnorr.cpp.o' failed
make[2]: *** [src/libCrypto/CMakeFiles/Crypto.dir/Schnorr.cpp.o] Interrupt
tests/Utils/CMakeFiles/Test_JoinableFunction.dir/build.make:62: recipe for target 'tests/Utils/CMakeFiles/Test_JoinableFunction.dir/Test_JoinableFunct                                               ion.cpp.o' failed
make[2]: *** [tests/Utils/CMakeFiles/Test_JoinableFunction.dir/Test_JoinableFunction.cpp.o] Interrupt
tests/Utils/CMakeFiles/Test_TimeLockedFunction.dir/build.make:62: recipe for target 'tests/Utils/CMakeFiles/Test_TimeLockedFunction.dir/Test_TimeLocke                                               dFunction.cpp.o' failed
make[2]: *** [tests/Utils/CMakeFiles/Test_TimeLockedFunction.dir/Test_TimeLockedFunction.cpp.o] Interrupt
tests/Utils/CMakeFiles/Test_Logger2.dir/build.make:62: recipe for target 'tests/Utils/CMakeFiles/Test_Logger2.dir/Test_Logger2.cpp.o' failed
make[2]: *** [tests/Utils/CMakeFiles/Test_Logger2.dir/Test_Logger2.cpp.o] Interrupt
CMakeFiles/Makefile2:1087: recipe for target 'src/libCrypto/CMakeFiles/Crypto.dir/all' failed
make[1]: *** [src/libCrypto/CMakeFiles/Crypto.dir/all] Interrupt
CMakeFiles/Makefile2:727: recipe for target 'src/depends/libDatabase/CMakeFiles/Database.dir/all' failed
make[1]: *** [src/depends/libDatabase/CMakeFiles/Database.dir/all] Interrupt
CMakeFiles/Makefile2:2729: recipe for target 'tests/Data/CMakeFiles/Test_CircularArray.dir/all' failed
make[1]: *** [tests/Data/CMakeFiles/Test_CircularArray.dir/all] Interrupt
CMakeFiles/Makefile2:4531: recipe for target 'tests/Utils/CMakeFiles/Test_DetachedFunction.dir/all' failed
make[1]: *** [tests/Utils/CMakeFiles/Test_DetachedFunction.dir/all] Interrupt
CMakeFiles/Makefile2:4379: recipe for target 'tests/Utils/CMakeFiles/Test_Serializable.dir/all' failed
make[1]: *** [tests/Utils/CMakeFiles/Test_Serializable.dir/all] Interrupt
CMakeFiles/Makefile2:4721: recipe for target 'tests/Utils/CMakeFiles/Test_SafeMath.dir/all' failed
make[1]: *** [tests/Utils/CMakeFiles/Test_SafeMath.dir/all] Interrupt
CMakeFiles/Makefile2:4927: recipe for target 'tests/Utils/CMakeFiles/Test_Logger2.dir/all' failed
make[1]: *** [tests/Utils/CMakeFiles/Test_Logger2.dir/all] Interrupt
CMakeFiles/Makefile2:4455: recipe for target 'tests/Utils/CMakeFiles/Test_TimeLockedFunction.dir/all' failed
make[1]: *** [tests/Utils/CMakeFiles/Test_TimeLockedFunction.dir/all] Interrupt
CMakeFiles/Makefile2:4417: recipe for target 'tests/Utils/CMakeFiles/Test_BoostBigNum.dir/all' failed
make[1]: *** [tests/Utils/CMakeFiles/Test_BoostBigNum.dir/all] Interrupt
CMakeFiles/Makefile2:4569: recipe for target 'tests/Utils/CMakeFiles/Test_JoinableFunction.dir/all' failed
make[1]: *** [tests/Utils/CMakeFiles/Test_JoinableFunction.dir/all] Interrupt
CMakeFiles/Makefile2:4645: recipe for target 'tests/Utils/CMakeFiles/Test_Logger1.dir/all' failed
make[1]: *** [tests/Utils/CMakeFiles/Test_Logger1.dir/all] Interrupt
CMakeFiles/Makefile2:4760: recipe for target 'tests/Utils/CMakeFiles/Test_ShardSizeCalculator.dir/all' failed
make[1]: *** [tests/Utils/CMakeFiles/Test_ShardSizeCalculator.dir/all] Interrupt
CMakeFiles/Makefile2:4799: recipe for target 'tests/Utils/CMakeFiles/Test_SWInfo.dir/all' failed
make[1]: *** [tests/Utils/CMakeFiles/Test_SWInfo.dir/all] Interrupt
CMakeFiles/Makefile2:3370: recipe for target 'tests/Message/CMakeFiles/Test_Messenger_Compatibility.dir/all' failed
make[1]: *** [tests/Message/CMakeFiles/Test_Messenger_Compatibility.dir/all] Interrupt
CMakeFiles/Makefile2:4683: recipe for target 'tests/Utils/CMakeFiles/Test_Logger3.dir/all' failed
make[1]: *** [tests/Utils/CMakeFiles/Test_Logger3.dir/all] Interrupt
Makefile:160: recipe for target 'all' failed
make: *** [all] Interrupt

**Pull request merged**